### DISCUSSION TITLE
Is it possible to have a single bundled JS file?

### DISCUSSION THREAD

**Discussion by ahiyalala** [parcel-bundler/parcel#8760] [Q&A]
Hello everyone,

Currently, when I run `parcel build` it generates an `index.xxx.js` file and also `component.xxx.js` and `anotherComponent.xxx.js` files which are separate scripts from my `src/scripts` that are loaded through `index.js`
```js
import("./components/component").then((component) => component.render());
import("./components/anotherComponent").then((component) => component.render());
```

Would it then be possible to bundle all JS files into one single JS file instead of having X number of minifed JS files?

**Comment by grimsteel** [comment]
It looks like you're using [dynamic imports](https://parceljs.org/features/code-splitting/#using-dynamic-imports) which load modules asynchronously - therefore, they are placed in separate files that can be loaded on demand.
If you wanted to generate one file, you could use a normal import

```js
import component from "./components/component";
import anotherComponent from "./components/anotherComponent";

// render it now or whenever you want to
component.render();
anotherComponent.render();
```

or CommonJS
```js
const component = require("./components/component");
const anotherComponent = require("./components/anotherComponent");
```

I hope this helps
